### DISCUSSION TITLE
How to use scripting in HFSS to export the exact coordinates of any element as a csv format?

### DISCUSSION THREAD

**Discussion by Zdaymen** [ansys/pyaedt#3911] [Q&A] [answered]
How to use scripting in HFSS to export the exact coordinates of any element as a csv format?

**Comment by Samuelopez-ansys** [comment, marked as answer]
@Zdaymen Not sure what you meant with exact coordinates, because for an sphere for instance, what is the exact coordinate?

What you can obtain with PyAEDT is the bounding box of each object (referred to the Global CS), the center, the mass... you just need to get the object:

`hfss.modeler.objects
`

You can get the list of objects names:
`names = hfss.modeler.objects.object_list`

And then get the first object information:
hfss.modeler[names [0]]

<img width="494" alt="image" src="https://github.com/ansys/pyaedt/assets/85613111/3dae376d-063d-4f18-ba9d-1bbca54283a5">

With this information, you will get the faces, vertices info... and finally you can export this information in a CSV using any python module like import csv:
https://stackoverflow.com/questions/45549424/exporting-python-output-into-csv-or-text-file-beginner

**Reply by mheroth** [reply]
Is it also possible to read out the coordinates that describe the edges? In my case, the z-coordinate is always zero and I'm looking for the coordinates that are shown to me via the measure tool in Electronics Desktop. The implemented vector graphics export has too much of a deviation from this for me.

**Reply by Samuelopez-ansys** [reply]
@mheroth If you debug, you will see how edges have all object edges with their information.
